shell/common/api/electron_api_native_image.cc

```diff
@@ -272,11 +281,11 @@ gin::Handle<NativeImage> NativeImage::Resize(v8::Isolate* isolate,
   if (width_set && !height_set) {
     // Scale height to preserve original aspect ratio
     size.set_height(width);
-    size = gfx::ScaleToRoundedSize(size, 1.f, 1.f / GetAspectRatio());
+    size = gfx::ScaleToRoundedSize(size, 1.f, 1.f / GetAspectRatio(args));
```

Comment:
I don't think this technique of passing `args` through works, because `GetScaleFactorFromOptions` calls `args->GetNext()`, so it'll be iterating through the arguments. This function will end up ultimately calling `args->GetNext()` twice.

Better to pull the scale factor out in each function which takes them, and then pass the scale factor along where necessary.